### ISSUE TITLE
prefer-primordials: Force `SafeArrayIterator` to be used in deno core

### ISSUE THREAD

**Issue by petamoriken** [denoland/deno_lint#968]
related: https://github.com/denoland/deno/issues/12614

### Lint Name

prefer-primordials

### Code Snippet

```ts
[1, 2, ...arr];
foo(1, 2, ...arr);
new Bar(1, 2, ...arr);
```

### Expected Result

Suggest using `SafeArrayIterator`.

```ts
const { SafeArrayIterator } = window.__bootstrap.primordials;
[1, 2, ...new SafeArrayIterator(arr)];
foo(1, 2, ...new SafeArrayIterator(arr));
new Bar(1, 2, ...new SafeArrayIterator(arr));
```

### Actual Result

Nothing.

### Additional Info

I think the following query (esquery) will help detect the problem points.

```css
:matches(ArrayExpression, CallExpression, NewExpression)
  > SpreadElement:not(
      [argument.type = NewExpression][argument.callee.name = SafeArrayIterator]
  )
```

### Version

deno 1.15.3 (release, aarch64-apple-darwin)
v8 9.5.172.19
typescript 4.4.2

**Comment by petamoriken** [comment]
The `for-of` loops also need to be handled.

```js
for (const val of new SafeArrayIterator(arr)) {}
```
